consumer-configuration/README.adoc

```diff
@@ -0,0 +1,347 @@
+////
+START GENERATED ATTRIBUTES
+WARNING: This content is generated by running npm --prefix .build run generate:attributes
+////
+
+
+:community:
+:imagesdir: ./images
+:product-version: 1
+:product-long: Application Services
+:product: App Services
+// Placeholder URL, when we get a HOST UI for the service we can put it here properly
+:service-url: https://cloud.redhat.com/beta/application-services/streams/
+:property-file-name: app-services.properties
+
+// Other upstream project names
+:samples-git-repo: https://github.com/redhat-developer/app-services-guides
+
+//URL components for cross refs
+:base-url: https://github.com/redhat-developer/app-services-guides/blob/main/
+:base-url-cli: https://github.com/redhat-developer/app-services-cli/tree/main/docs/
+:getting-started-url: getting-started/README.adoc
+:kafka-bin-scripts-url: kafka-bin-scripts/README.adoc
+:kafkacat-url: kafkacat/README.adoc
+:quarkus-url: quarkus/README.adoc
+:rhoas-cli-url: rhoas-cli/README.adoc
+:rhoas-cli-ref-url: commands
+:topic-config-url: topic-configuration/README.adoc
+:consumer-config-url: consumer-configuration/README.adoc
+
+////
+END GENERATED ATTRIBUTES
+////
+
+[id="chap-configuring-topics"]
+= Configuring consumer groups in {product-long}
+ifdef::context[:parent-context: {context}]
+:context: configuring-consumer-groups
+
+[IMPORTANT]
+====
+{product-long} is currently available for Development Preview. Development Preview releases provide early access to a limited set of features that might not be fully tested and that might change in the final GA version. Users should not use Development Preview software in production or for business-critical workloads. Limited documentation is available for Development Preview releases and is typically focused on fundamental user goals.
+====
+
+// Purpose statement for the assembly
+[role="_abstract"]
+Consumer groups share large data streams generated by producers from a given topic in a Kafka instance.
+Grouping consumers scales consumption to keep up with the rate of data produced.
+Consumers within a group don’t read data from the same partition, but can receive data from one or more partitions.
+
+As a developer of applications and services, you can view all the consumer groups that have access to a particular Kafka instance in {product}.
+If required, use the {product} web console to reset the offsets of consumer groups or delete consumer groups.
+
+As an alternative to using the {product} web console, you can use the `rhoas` command-line interface (CLI) or the Kafka `kafka-consumer-groups.sh` tool to review and update consumer groups.
+
+//Additional line break to resolve mod docs generation error
+
+[id="con-adding-consumer-group-id_{context}"]
+== How to add a consumer to a consumer group
```

Comment:
I think we should avoid "How to" headings because it's essentially a procedure that doesn't follow the template. If we want this to be a con, then could be a heading like "Consumer group ID" or "Consumers in consumer groups". But then we're suppressing the most important question of the doc, which is how to add consumers, as you've said. 

So if it were me, for clarity and searchability while still complying with FCC templates, I'd make it a simple proc, "Adding a consumer to a consumer group". Then you basically keep the same intro content and make it a single-step proc with the needed info.

Something like this. Note that I've also removed some ital that I thought was unneeded, reordered some things a little that made it clearer to me, and tweaked a couple words. Feel free to take or disregard what you'd like. And I believe the latest FCC guideline for single-step procs is to bullet it, whereas before it was to be just empty. So I've bulletted accordingly.

```
[id="proc-adding-consumer-group-id_{context}"]
== Adding a consumer to a consumer group

A group ID is required for a consumer to be able to join a consumer group. The group ID is the name of the consumer group. When you configure a consumer client application to access Kafka, you assign a group ID to associate it with a consumer group. All consumers with the same group ID belong to the consumer group with that ID. Consumer groups connected to a Kafka instance are displayed in the {product} web console. 

.Procedure
* Add the consumer group ID as a property to your consumer configuration. The group ID can't be null or empty.
+
--
.Example group ID property configuration for a consumer
[source,properties]
----
group.id=my-group-id
----

.Property attributes
[cols="25%,75%"]
|===

h|Type
|String

h|Default value
|None

h|Supported values
|Letters (Aa-Zz), numbers (0-9), underscores ( _ ), or hyphens ( - )
|===

If you're using Kafka bin scripts to consume messages, you can use the `kafka-console-consumer.sh` tool to create a group ID:
[source,subs="+quotes,+attributes"]
----
bin/kafka-console-consumer.sh --bootstrap-server __<bootstrap_server>__ --topic test-topic --group my-consumer-group
----

The `<bootstrap_server>` is the bootstrap server endpoint for your service account.
You copy this information for the Kafka instance in the {product} web console by selecting the options menu (three vertical dots) and clicking *View connection information*.

[NOTE]
====
If you add more consumers than the number of partitions in a consumer group, the additional consumers don't help throughput but they function as standby consumers that can replace any consumers that stop functioning. The fewer consumers you use in order to meet throughput, the more resources you save.
====

--

.Additional resources
* {base-url}{getting-started-url}[_Getting started with {product}_^]
* https://kafka.apache.org/documentation/#consumerconfigs[Consumer Configs^] in Kafka documentation
```

Reply:
Thanks Stetson. I didn't think it merited a procedure, but your suggestion works. Applied.